src/02-room-details.js

```diff
@@ -49,7 +75,34 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  let list = [];
+  let finalList = [];
+  let eachRoomId = "";
+
+  // Check if input id is found in the rooms array
+  for (const room of rooms) {
+    if (room.roomId === id) {
+      // if input 'id' is present in rooms
+      list = room.connectsTo;
+    }
+  }
+
+  for (const i of list) {
+    eachRoomId = i;
+    for (const i2 of rooms) {
+      if (i === i2.roomId) {
+        finalList.push(i2.name);
+      }
+    }
+  }
+
+  if (eachRoomId === "" || eachRoomId === "incorrect-id")
```

Comment:
Hello Angie, thank you for your feedback!!
I am a bit confused. My understanding was that the test was specifically asking for 'incorrect-id' .

When I ran the test,  this was what I received from i-Terminal ->
Expected: "Room with ID of 'incorrect-id' could not be found."
Received  "Room with ID of 'xwG7O4wQl' could not be found." 
and because it wasn't specifically returning 'incorrect-id', it was not passing the test.